nestkernel/connector_model.h

```diff
@@ -199,7 +219,19 @@ class GenericConnectorModel : public ConnectorModel
   SecondaryEvent*
   get_secondary_event() override
   {
-    return default_connection_.get_secondary_event();
+    constexpr bool is_primary = flag_is_set( ConnectionT::properties, nest::ConnectionModelProperties::IS_PRIMARY );
+    constexpr bool has_get_secondary_event = has_get_secondary_event_t< ConnectionT >::value;
+    static_assert(
+      is_primary xor has_get_secondary_event, "Non-primary connections have to provide get_secondary_event()" );
+    if constexpr ( ( not is_primary ) and has_get_secondary_event )
```

Comment:
should be sufficient to only check one of them after the assert

Reply:
Should be sufficient, but also doesn't hurt to check for both at compile time, right?